src/classes/BDI_DataImportService.cls

```diff
@@ -704,62 +704,116 @@ global with sharing class BDI_DataImportService {
+        Map<String, String> parametersByName = new Map<String, String>();
         Map<String, String> dataImportFieldToTargetField = new Map<String, String>();
+        Map<String, Set<String>> targetFieldsBySource = new Map<String, Set<String>>();
+        Boolean useCustomMetadataForFieldMapping = false;
+        if (bdiSettings.MapWithCustomMetadataTypes__c) {
+            List<BDI_Field_Mapping__mdt> bdiFIeldMappings = [SELECT
+                                                                    Source_Field__c,
+                                                                    Target_Field__c
+                                                             FROM
+                                                                    BDI_Field_Mapping__mdt];
 
-        for (String dataImportField : dataImportFields) {
+            for (BDI_Field_Mapping__mdt customMetadata : bdiFIeldMappings) {
+                String sourceField = customMetadata.Source_Field__c.toLowerCase();
+                if(targetFieldsBySource.containsKey(sourceField)) {
+            		Set<String> targetFields = targetFieldsBySource.get(sourceField);
+            		targetFields.add(customMetadata.Target_Field__c);
+                    targetFieldsBySource.put(sourceField, targetFields);
+	            } else {
+            		targetFieldsBySource.put(sourceField, new Set<String> { customMetadata.Target_Field__c });
+            	}
+            }
+
+            useCustomMetadataForFieldMapping = true;
+        }
 
-            DescribeFieldResult fieldDescribe =
-                UTIL_Describe.getFieldDescribe(UTIL_Namespace.StrTokenNSPrefix('DataImport__c'), dataImportField);
-            String helpText = fieldDescribe.getInlineHelpText();
-
-            if (helpText != null && helpText != '') {
-                List<String> parts = helpText.split('\\.', 2);
-
-                if (parts.size() == 2 && parts[0] == dataImportObjectName && parts[1] != '') {
-                    String fieldName = parts[1];
-                    //if we aren't in the npsp namespace, remove those tokens from help text
-                    //also removes tokens from actual detected namespace, if we happen to be
-                    //in a non-npsp namespace
-                    if (UTIL_Namespace.getNamespace() != 'npsp') {
-                        String fieldNameOld = fieldName;
-                        fieldName = UTIL_Namespace.StrTokenRemoveNSPrefix(fieldName);
-
-                        //if we have a detected namespace, and removed a namespace from the current field
-                        //add the current namespace back in, allowing this code to run in any namespace
-                        if (fieldName != fieldNameOld && UTIL_Namespace.getNamespace() != '') {
-                            fieldName = UTIL_Namespace.StrAllNSPrefix(fieldName);
-                        }
+        for (String dataImportField : dataImportFields) {
+            DescribeFieldResult fieldDescribe;
```

Comment:
Should this be dropped into the else block, or is there a potential need for it above?

Reply:
No, this line made sense here before I moved the logic to the auxiliar method because it was called bellow for booth cases, but it is not necessary now.

Reply:
I am removing that line.